### DISCUSSION TITLE
getting undefined symbols when running tests

### DISCUSSION THREAD

**Discussion by macinsmith** [pointfreeco/swift-composable-architecture#2941] [Q&A]
I'm using the Composable Architecture Tutorial. (I'm using 1.9.2 of Composable Architecture, Xcode 15.2, iOS target 17.2)

1. The app builds and runs correctly.
2. Created a unit test target
3. Added CounterFeatureTests.swift and added it to the test target
4. Added the func testCounter() to the CounterFeatureTests.swift and included Composable Architecture
5. Added Composable Architecture to the test target
6. @testable import  CounterFeature in the test.
7. cmd/U to run the test (or clicked on diamond by test)
8. Gave me a failure.  Opening the log I found 
9. `Link CounterFeature (arm64)`
10. `Could not find or use auto-linked framework 'CoreAudioTypes': framework 'CoreAudioTypes' not found`
11. `Undefined symbols:`
12. `Linker command failed with exit code 1 (use -v to see invocation)`
13. Added import for CoreAudioTypes, but didn't help (I didn't think it would)

Without adding the test target cmd/U would not work, nor were there diamonds by the testCounter func

**Comment by mbrandonw** [comment]
Hi @macinsmith, you shouldn't need to do this step:

> Added Composable Architecture to the test target

Your app target depends on TCA, and your test target depends on the app target. That should be enough.

Most likely there is just something misconfigured in your project. Is there a reason you created a unit test target from scratch? Typically the Xcode template creates a test target for you automatically. I would try creating a fresh Xcode project from scratch and see if it works.

And by the way, the "CoreAudioTypes" part is just a warning, it is not the root cause of the problem.

**Comment by macinsmith** [comment]
Thanks for your suggestions.

I rebooted my machine, created a new project with testing enabled, and just @testable to the app target and it finally worked.

It is working now  on my M1 Mac, running Sonoma and Xcode 15.2, ComposableArchitecture 1.9.2, iOS 17.2 in simulator

**Comment by macinsmith** [comment]
Thanks for your suggestions.

I rebooted my machine, created a new project with testing enabled, and just @testable to the app target and it finally worked.

It is working now  on my M1 Mac, running Sonoma and Xcode 15.2, ComposableArchitecture 1.9.2, iOS 17.2 in simulator

On Mar 22, 2024, at 5:31 PM, Brandon Williams ***@***.***> wrote:


Hi @macinsmith <https://github.com/macinsmith>, you shouldn't need to do this step:

Added Composable Architecture to the test target

Your app target depends on TCA, and your test target depends on the app target. That should be enough.

Most likely there is just something misconfigured in your project. Is there a reason you created a unit test target from scratch? Typically the Xcode template creates a test target for you automatically. I would try creating a fresh Xcode project from scratch and see if it works.

And by the way, the "CoreAudioTypes" part is just a warning, it is not the root cause of the problem.
